### PR TITLE
Add test-utils submodule (1/2)

### PR DESCRIPTION
**Pull request by Pessimistress**
See documentation in PR

#### Change List
- Add `@luma.gl/test-utils` submodule

### CODE REVIEW

File: docs/api-reference/test-utils/test-render.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+```js
+const {TestRenderer} = require('@luma.gl/test-utils');
+const {Cube} = require('@luma.gl/core');
+
+const TEST_CASES = [
+  {
+    name: 'Render A Cube',
+    onRender: ({gl, done}) => {
+      const model = new Cube(gl);
+      model.draw();
```
Comment (ibgreen) [inline docs/api-reference/test-utils/test-render.md]: Non-working example? Is some uniform needed? At least add `draw(...)` to indicate incompleteness?

File: docs/api-reference/test-utils/test-render.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  .add(TEST_CASES)
+  .run({
+    onTestFail: window.browserTestDriver_fail
+  })
+  .finally(window.browserTestDriver_finish);
+```
+
+## Method
+
+### constructor
```
Comment (ibgreen) [inline docs/api-reference/test-utils/test-render.md]: Nit: `constructor(props : Object)`

File: docs/api-reference/test-utils/test-render.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+In your script that is run on the browser:
+
+```js
+const {TestRenderer} = require('@luma.gl/test-utils');
+const {Cube} = require('@luma.gl/core');
+
+const TEST_CASES = [
+  {
+    name: 'Render A Cube',
+    onRender: ({gl, done}) => {
```
Comment (ibgreen) [inline docs/api-reference/test-utils/test-render.md]: Where is the context coming from? Argument to `new TestRenderer`?

File: docs/api-reference/test-utils/test-render.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  - `done` (Function) - must be called when the test case is done rendering and ready for screen capture and comparison.
+* `onFinalize` (Function) - called once when the test case is done to finalize all resources. Receives a single object that is the [AnimationLoop callback parameters](/docs/api-reference/core/animation-loop.md#callback-parameters).
+
+### run(options: Object)
+
+Run all test cases.
+
+Parameters:
+
+* `timeout` (Number) - time to wait for each test case to resolve (by calling the `done` callback) before aborting, in milliseconds. Default `2000`.
```
Comment (ibgreen) [inline docs/api-reference/test-utils/test-render.md]: Possibly `options.timeout`...

File: docs/api-reference/test-utils/test-render.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+* `imageDiffOptions` (Object) - image diffing options for all test cases. This will be overridden if a test case defines its own `imageDiffOptions`. See "Image Diff Options" section below.
+* `onTestStart` (Function) - callback when a test starts. Receives the current test case. Default logs the test name to console.
+* `onTestPass` (Function) - callback when a test passes. Receives the current test case and the diffing result. Default logs the pixel matching percentage to console.
+* `onTestFail` (Function) - callback when a test fails, either because the matching rate is below threshold or a critical error. Receives the current test case. Default logs the error message or the pixel matching percentage to console.
+
+Returns: a `Promise` that resolves when all test cases are done.
+
+
+## Member
```
Comment (ibgreen) [inline docs/api-reference/test-utils/test-render.md]: Members?

File: modules/test-utils/README.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
+# @luma.gl/test-utils
```
Comment (ibgreen) [inline modules/test-utils/README.md]: I see the value in having info in the npm readme page, but In general I would avoid duplicating information, in the past similar setups have quickly gotten out of sync. A short prologue and then linking to the docs is the right approach IMHO.

**Comment by ibgreen** [comment]
@georgios-uber this should be perfect for your glTF conformance tests...

File: docs/api-reference/test-utils/test-render.md (hunk 3 above)
Comment (Pessimistress) [reply docs/api-reference/test-utils/test-render.md]: The context comes from AnimationLoop. I'll add a comment.

**Pull request merged**